backend/corpora/dataset_processing/process.py

```diff
@@ -212,14 +211,23 @@ def main():
         cancel_dataset(dataset_id)
     except (ValidationFailed, ProcessingFailed, ConversionFailed) as e:
         (status,) = e.args
-        if is_last_attempt:
-            update_db(dataset_id, processing_status=status)
+        update_db(dataset_id, processing_status=status)
         logger.exception("An Error occurred while processing.")
         return_value = 1
     except Exception as e:
         (status,) = e.args
-        if is_last_attempt and isinstance(status, dict):
+        if isinstance(status, dict):
             update_db(dataset_id, processing_status=status)
+        else:
+            if step_name == "download-validate":
```

Comment:
Can you add some test cases to verify this behavior? I recently update the test code to be easier to write tests for this area https://github.com/chanzuckerberg/single-cell-data-portal/blob/b1218c03bb35f997e61568104095d95272d5ce0c/tests/unit/processing_container/test_process.py#L12